lib/Platform/Logging.cpp

```diff
@@ -36,12 +59,25 @@ void hermesLog(const char *componentName, const char *fmt, ...) {
   int numCharsWritten = vsnprintf(buffer.get(), numChars + 1, fmt, args);
   (void)numCharsWritten;
   // Log all messages as public here. Hermes will never log private information.
+#if defined(__APPLE__)
+  static os_log_t hermesLogger = os_log_create("dev.hermesengine", "Default");
   os_log_with_type(
       hermesLogger,
       OS_LOG_TYPE_INFO,
       "%{public}s: %{public}s",
       componentName,
       buffer.get());
+#elif defined(_MSC_VER)
+  static std::once_flag g_initEtw_once;
+  std::call_once(
+      g_initEtw_once, []() { TraceLoggingRegister(g_hTraceLoggingProvider); });
+
+  TraceLoggingWrite(
```

Comment:
If `TraceLoggingWrite` has a version that takes varargs like `vfprintf` below, you can save some malloc calls in the `va_copy` path here.
If not, this is fine.

Reply:
TraceLoggingWrite takes varargs .. but what we need is a version of TraceLoggingString that takes varargs, which i can't find so far :(